components/eventing-controller/pkg/backend/jetstreamv2/jetstream.go

```diff
@@ -297,31 +299,37 @@ func (js *JetStream) cleanupUnnecessaryJetStreamSubscribers(jsSub Subscriber,
 		"Deleting JetStream subscription because it was deleted from subscription types",
 		"jetStreamSubject", info.Config.FilterSubject,
 	)
-	return js.deleteSubscriptionFromJetStream(jsSub, key, log)
+	return js.deleteSubscriptionFromJetStream(jsSub, key)
 }
 
-// deleteSubscriptionFromJS deletes subscription from JetStream and from in-memory db.
-func (js *JetStream) deleteSubscriptionFromJetStream(jsSub Subscriber,
-	jsSubKey SubscriptionSubjectIdentifier, log *zap.SugaredLogger) error {
+// deleteSubscriptionFromJetStream deletes subscription from NATS server and from in-memory db.
+func (js *JetStream) deleteSubscriptionFromJetStream(jsSub Subscriber, jsSubKey SubscriptionSubjectIdentifier) error {
 	// unsubscribe call to JetStream is async hence checking the status of the connection is important
 	if err := js.checkJetStreamConnection(); err != nil {
 		return err
 	}
 
+	if err := js.unsubscribeOnNats(jsSub, jsSubKey); err != nil {
```

Comment:
Maybe rename it to `deleteSubscriptionWithConsumer`, otherwise I see no difference between `deleteSubscriptionFromJetStream` and `unsubscribeOnNats`.

Reply:
Check out the new version